runtime/lua/vim/lsp/util.lua

```diff
@@ -514,7 +514,8 @@ function M.convert_input_to_markdown_lines(input, contents)
 
       -- Some servers send input.value as empty, so let's ignore this :(
       -- assert(type(input.value) == 'string')
-      list_extend(contents, split_lines(input.value or ''))
+      local stripped_message = string.gsub(input.value or '', "\\", "")
```

Comment:
What if there's a valid `\`  that's not used to escape special characters?

Reply:
Then it will be stripped, but to be honest I don't know what else to do about that. I'm betting the frequency of unintended rendered escapes >> frequency of intended \

Reply:
Look at the [link](https://github.com/golang/tools/blob/7e51fbd4c416938b4e68e04220e3d47bf6ec33d2/internal/lsp/source/comment.go#L93-L106) provided by @owarai aboved, there aren't too much characters that get escaped.
I think we could make a dictionary of them instead. Something like:
```lua
local unescapedMap = {
  ['\\\\'] = '\\',
  ['\\`'] = '`',
  ['\\*'] = '*',
-- etc.
}

local stripped_message = input.value or ''
for k,v in pairs(unescapedMap) do
   stripped_message = string.gsub(stripped_message, k, v)
end
```
(don't know if that's correct lua code)

Or we can use some regex magic:
```
'\\([\\`*{}[\]()#+\-.!_>~|"$%&'\/:;<=?@^])'  -->  '$1'
```
(cmiir, I don't know regex in lua)

Reply:
I did, in fact, look at their link. 

Lua doesn't have a regex library (a full regex library would be larger than LuaJIT's entire source code).

If you're interested, you can find details about lua pattern matching here: https://www.lua.org/manual/5.1/manual.html#5.4.1

Reply:
We have `vim.regex`

Reply:
Oh, I'm aware, I was just responding to "I don't know regex in lua", I don't think we need a regex for this though since I updated it to match any character occuring after \